modules/siddhi-extensions/input-transports/jms-input-transport/src/main/java/org/wso2/siddhi/extension/input/transport/jms/util/JMSListener.java

```diff
@@ -0,0 +1,192 @@
+                    if (stm.getConsumerCount() > 0) {
+                        log.info("Started to listen on destination : " + stm.getDestinationJNDIName() +
+                                " of type " + JMSUtils.getDestinationTypeAsString(stm.getDestinationType()) +
+                                " for listener " + listenerName);
+                        return;
+                    }
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException ignore) {
+                    }
+                }
+
+                log.warn("Polling tasks on destination : " + stm.getDestinationJNDIName() +
+                        " of type " + JMSUtils.getDestinationTypeAsString(stm.getDestinationType()) +
+                        " for listener " + listenerName + " have not yet started after 3 seconds ..");
+            } else {
+                log.error("Unable to continue server startup as it seems the JMS Provider " +
+                        "is not yet started. Please start the JMS provider now.");
+                retryDuration = (long) (retryDuration * reconnectionProgressionFactor);
+                log.error("Connection attempt : " + (r++) + " for JMS Provider failed. Next retry in "
+                        + (retryDuration / 1000) + " seconds");
+                if (retryDuration > maxReconnectDuration) {
+                    retryDuration = maxReconnectDuration;
+                }
+                try {
+                    Thread.sleep(retryDuration);
+                } catch (InterruptedException ignore) {
+                }
+            }
+        }
+    }
+
+    private boolean checkJMSConnection(JMSTaskManager stm) {
+
+        Connection connection = null;
+        boolean connectionEstablished = false;
+        Hashtable<String, String> jmsProperties = stm.getJmsProperties();
+        try {
+            ConnectionFactory jmsConFactory = null;
+            try {
+                jmsConFactory = JMSUtils.lookup(
+                        new InitialContext(stm.getJmsProperties()), ConnectionFactory.class,
+                        stm.getConnFactoryJNDIName());
+            } catch (NamingException e) {
+                log.error("Error looking up connection factory : " + stm.getConnFactoryJNDIName() +
+                        "using JNDI properties : " + jmsProperties, e);
+            }
+            connection = JMSUtils.createConnection(
+                    jmsConFactory,
+                    jmsProperties.get(JMSConstants.PARAM_JMS_USERNAME),
+                    jmsProperties.get(JMSConstants.PARAM_JMS_PASSWORD),
+                    stm.isJmsSpec11(), stm.isQueue(), stm.isSubscriptionDurable(), stm.getDurableSubscriberClientId());
+            connectionEstablished = connection != null;
+        } catch (JMSException ignore) {
+            // we silently ignore this as a JMSException can be expected when connection is not available
```

Comment:
There can be various other reasons like authentication failure etc. as well right? If this is ignored, how to find out the issue?